gap/attributes/attr.gi

```diff
@@ -774,26 +774,75 @@ function(G, x)
 end);
 
 InstallMethod(InversesOfSemigroupElementNC,
-"for a semigroup and a multiplicative element",
-[IsSemigroup, IsMultiplicativeElement],
+"for a semigroup that can use froidure-pin and a multiplicative element",
+[CanUseFroidurePin, IsMultiplicativeElement],
+function(S, a)
+  local R, L, inverses, e, f, s;
+  R := RClass(S, a);
+  L := LClass(S, a);
+  inverses := EmptyPlist(NrIdempotents(R) * NrIdempotents(L));
+
+  for e in Idempotents(R) do
+    s := RightGreensMultiplierNC(S, a, e) * e;
+    for f in Idempotents(L) do
+      Add(inverses, f * s);
+    od;
+  od;
+  return inverses;
+end);
+
+InstallMethod(InversesOfSemigroupElement,
+"for a semigroup that can use froidure-pin and a multiplicative element",
+[CanUseFroidurePin, IsMultiplicativeElement],  # to beat the library method
 function(S, x)
   if not IsFinite(S) then
     TryNextMethod();
+  elif not x in S then
+    ErrorNoReturn("the 2nd argument (a mult. element) must belong to the 1st ",
+                  "argument (a semigroup)");
   fi;
-  return Filtered(EnumeratorSorted(S), y -> x * y * x = x and y * x * y = y);
+  return InversesOfSemigroupElementNC(S, x);
 end);
 
-InstallMethod(InversesOfSemigroupElement,
+InstallMethod(OneInverseOfSemigroupElementNC,
 "for a semigroup and a multiplicative element",
-[IsSemigroup, IsMultiplicativeElement], 1,  # to beat the library method
+[IsSemigroup, IsMultiplicativeElement],
+function(S, x)
+    if not IsFinite(S) then
+        TryNextMethod();
+    fi;
+    return First(EnumeratorSorted(S),
+    y -> x * y * x = x and y * x * y = y);
+end);
+
+InstallMethod(OneInverseOfSemigroupElementNC,
+"for CanUseFroidurePin and a multiplicative element",
+[CanUseFroidurePin, IsMultiplicativeElement],
+function(S, a)
+  local R, L, e, f, s;
+  R := RClass(S, a);
+  L := LClass(S, a);
+  e := Idempotents(R);
+  if IsEmpty(e) then
+    return fail;
+  fi;
+  e := e[1];
+  s := RightGreensMultiplierNC(S, a, e);
+  f := Idempotents(L)[1];
+  return  f * s * e;
+end);
+
+InstallMethod(OneInverseOfSemigroupElement,
+"for a semigroup and a multiplicative element",
+[IsSemigroup, IsMultiplicativeElement],
 function(S, x)
   if not IsFinite(S) then
```

Comment:
Can you please add a comment here that we don't `TryNextMethod` because the next method will run forever.